//===-- logon.proto - logon request & response message payloads -----------===//
//
// Copyright (c) 2025 Nuclearist <nuclearist@teknology-hub.com>
// Part of tek-steamclient, under the GNU General Public License v3.0 or later
// See https://github.com/teknology-hub/tek-steamclient/blob/main/COPYING for
//    license information.
// SPDX-License-Identifier: GPL-3.0-or-later
//
//===----------------------------------------------------------------------===//
syntax = "proto3";

package tek.steamclient.cm.msg_payloads;

option optimize_for = LITE_RUNTIME;

// Request message payload for EMSG_CLIENT_LOG_ON.
message LogonRequest {
  // Steam CM protocol version.
  optional uint32 protocol_version = 1;
  // Client language name.
  optional string client_language = 6;
  // Client OS type.
  optional int32 client_os_type = 7;
  // If not set, the token may be invalidated on logoff.
  optional bool should_remember_password = 8;
  // Account authentication token.
  optional string access_token = 108;
}

// Response message payload for EMSG_CLIENT_LOG_ON_RESPONSE.
message LogonResponse {
  // Logon result code.
  optional int32 eresult = 1;
}
